### PR TITLE
Include Non Exported CardDefs and FieldDefs to LHS (in this file) panel

### PR DESCRIPTION
**Pull request by tintinthong**
This PR supports non-exports of card and field definitions. Previously when you clicked on a card defined locally, the inheritance panel will not pop up -- now it should

https://github.com/cardstack/boxel/assets/8165111/b7491a1a-7b76-4156-8175-b97a90524006

Here are a few good files in the drafts-realm to look at: 

- in-this-file.gts
Screenshot 2023-10-10 at 6 13 56 PM

- vendor.gts
Screenshot 2023-10-10 at 6 11 57 PM

- payment-method.gts
Screenshot 2023-10-10 at 6 12 43 PM

- asset.gts
Screenshot 2023-10-10 at 6 13 16 PM

- contact-card.gts

Screenshot 2023-10-10 at 6 14 20 PM

### CODE REVIEW

File: packages/host/app/components/operator-mode/detail-panel.gts
Comment (tintinthong) [inline packages/host/app/components/operator-mode/detail-panel.gts]: PS. I intentionally changed this for ease of review. Its is one of those things I think may be easier to consider when we handle the 

https://linear.app/cardstack/issue/CS-5993/handle-situation-where-each-definition-has-the-same-name

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (tintinthong) [inline packages/host/app/components/operator-mode/code-mode.gts]: The purpose of this logic here is to really consolidate the information that we obtain from 
- moduleSyntax. The elements derive using the babel plugin
- importedModule. The module resource that collects exports of cards or fields

The point of it is that we can't just simply know that an element is a card or field or not from moduleSyntax. Rather, we have to look into the relationships of the cards or fields received from the importedModule. In doing so, we can infer about other cards or fields that are defined locally but not exported.

Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: Ive refactored this so its more readable

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [inline packages/host/app/components/operator-mode/code-mode.gts]: if it's `private` there is no need to use `_` prefix to indicate that this is not part of the public API

Comment (habdelra) [reply packages/host/app/components/operator-mode/code-mode.gts]: if you really wanna make it hands off, then use `#selectedElement`

Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: I meant to actually rename the selectedElement using an underscore since im running out of names and I didn't want to use something like `selectedElementInFile`

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [inline packages/host/app/components/operator-mode/code-mode.gts]: can we move this resource into it's own module in order to reduce code-mode.ts's size? (I couldn;t make a comment at the beginning of the resource so i made it here)

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [inline packages/host/app/components/operator-mode/code-mode.gts]: it would be good to move this out of the code-mode module--it can live in the same module as the resource to which this pertains (which can be moved out of this module too  from my comment above)

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [reply packages/host/app/components/operator-mode/code-mode.gts]: Also, i'd suggest that we move any other inline resources to their own modules (this is simple way to break down this module into smaller pieces)

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [inline packages/host/app/components/operator-mode/code-mode.gts]: i always use `console.error()` to log errors so they appear in red in the console

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [inline packages/host/app/components/operator-mode/code-mode.gts]: oh i see this is why the property `selectedElement` became `_selectedElement`

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [reply packages/host/app/components/operator-mode/code-mode.gts]: at this point code-mode is so long that we should endeavor to not add anymore new lines to it. if you need to add anything to it, make sure to remove something from it.

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (jurgenwerk) [inline packages/host/app/components/operator-mode/code-mode.gts]: Would combinations of early returns and `continue`s help taming the big nesting depth of this function?

**Review by jurgenwerk — COMMENTED** [review]
I agree that code mode component code got a bit out of hand and and this point we should perhaps commit to making it smaller, not bigger. Code wise this looks ok to me, just a minor comment. I didn't test it yet manually but planning on doing that soon.

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: https://github.com/cardstack/boxel/pull/716/commits/cbf2d18997481f7f196de22b36728117ca52e52f

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: https://github.com/cardstack/boxel/pull/716/commits/3534856782445a647913007b0b837c178c129b2c. I didn't use continue but I did use early return

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (habdelra) [inline packages/host/app/components/operator-mode/code-mode.gts]: we talked about a better name here. please make sure to change that before merging

**Comment by habdelra** [comment]
Based on our pairing today, are you blocked on this pending a discussion around how we want to proceed regarding module mutations and loader sandboxing?

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: Fixed here https://github.com/cardstack/boxel/pull/716/commits/380cf48e1d2e1119804a4ef8b3bcab0fea8f7036

Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: Gist of it is 
- InThisFileResource -> ModuleContentsResource
- elements -> declarations

**Comment by tintinthong** [comment]
> Based on our pairing today, are you blocked on this pending a discussion around how we want to proceed regarding module mutations and loader sandboxing?

I would consider our discussion of "sandboxing" outside scope of this pr. Therefore, not blocked

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (jurgenwerk) [inline packages/host/app/components/operator-mode/code-mode.gts]: Typo

File: packages/host/app/resources/module-contents.ts
Comment (jurgenwerk) [inline packages/host/app/resources/module-contents.ts]: Looks like something is missing in the comment - an already what?

**Review by jurgenwerk — APPROVED** [review]
Pointed out some minor typos. Looks much better organized after refactoring and renaming. Looks good to me but is it possible to include a test for this?

File: packages/host/app/components/operator-mode/code-mode.gts
Comment (tintinthong) [reply packages/host/app/components/operator-mode/code-mode.gts]: https://github.com/cardstack/boxel/pull/716/commits/4e4e226621f0389cdfd391b927955ddbfeb79212

File: packages/host/app/resources/module-contents.ts
Comment (tintinthong) [reply packages/host/app/resources/module-contents.ts]: https://github.com/cardstack/boxel/pull/716/commits/798bb26b54cce6777a296f49c3dd321ddfe6d761 

This was poor english

**Comment by tintinthong** [comment]
> Pointed out some minor typos. Looks much better organized after refactoring and renaming. Looks good to me but is it possible to include a test for this?

There is a test already [here](https://github.com/cardstack/boxel/blob/6d1eafc3961dfcb4409d33d89804454f2090e50f/packages/host/tests/acceptance/code-mode-test.ts#L893-L983) which should model all schenarios

**Pull request merged**
